### ISSUE TITLE
[Security] Support for strict CSP

### ISSUE THREAD

**Issue by knalltuete86** [bpatrik/pigallery2#671]
**Is your feature request related to a problem? Please describe.**
Defend against Cross-Site-Scripting (CSS).
Using an Reverse Proxy (e.g. nginx) before Pigallery2 and hardening it by using Content-Security-Policies (CSP) like described [here](https://web.dev/strict-csp/ ) is currently not possible as far as I understand.

Main blocking points:
- unsafe inline scripts without nonces
- unsafe inline styles
- dynamic generated scripts with dynamic hashes

**Describe the solution you'd like**
Adding Support for strict CSP by NPM (like this: https://www.npmjs.com/package/strict-csp)

**Describe alternatives you've considered** (optional)
Adding nonces to all scripts and styles and publish them in order to solve this within Configuration of Reverse Proxy.
(hashes seem not to be suitable due to dynamic generated content).

**Comment by bpatrik** [comment]
Using reverse proxy is highly recommended. The default docker compose setup also contains it.

i will need to look into csp.

**Comment by knalltuete86** [comment]
Yes I'm aware of using reverse proxy for all my self-hosted services. The main purpose for me to use PiGallery is to share photos & videos with family & friends. 
Unfortunately I cannot secure all clients with strong authentication. That's why I'm afraid exposing it to the internet while CSP issues cannot be fixed.

I guess for most users who use it on local network or by VPN only it doesn't matter. So may it can be fixed with small changes via reverse procxy configuration.

Regards,
Horst
